### PR TITLE
feat: Add sorting on alias/name column

### PR DESCRIPTION
**Pull request by banusha2**
## Context
To add sorting on alias/name column in Dashboard.

## Objective
1.Sort based on alias if pipeline has alias, in alphabetical order.
2.Sort based on pipeline name if alias not exist, in alphabetical order.

## References
https://github.com/screwdriver-cd/screwdriver/issues/2765

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/components/collection-view/template.hbs [hunk 1]
```diff
@@ -149,11 +149,20 @@
           <thead>
             <tr>
               <th class="collection-pipeline__choose"></th>
-              <th class="app-id">
+              <th class="app-id" onclick={{action "setSortBy" (if (eq (get sortBy 0) (if pipeline.settings.aliasName "pipeline.settings.aliasName:desc" "scmRepo.name:desc")) (if pipeline.settings.aliasName "pipeline.settings.aliasName:asc" "scmRepo.name:asc") (if pipeline.settings.aliasName "pipeline.settings.aliasName:desc" "scmRepo.name:desc"))}}>
```
Comment (adong) [inline app/components/collection-view/template.hbs:152]: Wow, this logic probably is correct and confusing. how about convert all 

```
(if (eq (get sortBy 0) (if pipeline.settings.aliasName "pipeline.settings.aliasName:desc" "scmRepo.name:desc")) (if pipeline.settings.aliasName "pipeline.settings.aliasName:asc" "scmRepo.name:asc") (if pipeline.settings.aliasName "pipeline.settings.aliasName:desc" "scmRepo.name:desc"))}
```

these to a computed property, as

```
onclick={{action "setSortBy" aliasName}}

aliasName: computed('pipeline.settings.aliasName',  'scmRepo.name', 'sortBy') {
	const sortingCriteria = '';
	if () {
   sortingCriteria = 'xx';
  } else {
    sortingCriteria = 'yy'
  }
  return sortingCriteria;
}

```

**Review by adong — CHANGES_REQUESTED** [review]

File: app/components/collection-view/component.js [hunk 2]
```diff
@@ -155,6 +155,28 @@ export default Component.extend({
...
+  aliasNameSort: computed(
+    'pipeline.settings.aliasName',
+    'scmRepo.name',
+    'sortBy',
+    function () {
+      let sortingCriteria = '';
+
+      if (this.hasAliasName) {
+        sortingCriteria =
+          this.sortBy[0] === 'pipeline.settings.aliasName:desc'
```
Comment (adong) [inline app/components/collection-view/component.js:167]: Can we use plain if else, and not use ternary operator `?` ?

**Comment by adong** [comment]
Also please add test cases, pipeline collection dashboard with: 

1) pipeline (without alias) default list
1) pipeline (without alias) list ascending order
1) pipeline (without alias) list descending order

1) pipeline (with alias) default list
1) pipeline (with alias) sorted list ascending order
1) pipeline (with alias) sorted list descending order

There may be a duplicate between 1) & 4), since default sorting criteria would be the same, and based on added time (if I recall correctly), still I list all possible scenarios I can think of.

**Review by jithine — APPROVED** [review]

**Comment by tkyi** [comment]
Still missing test cases listed here: https://github.com/screwdriver-cd/ui/pull/830#issuecomment-1262542043

**Comment by banusha2** [comment]
Feature will be served using PR #855

**Pull request closed without merging**